app/services/postcode_validator.rb

```diff
@@ -0,0 +1,9 @@
+class PostcodeValidator
+  VALID_POSTCODE = /^([A-Z]{1,2}[0-9][A-Z0-9]? ?[0-9][A-Z]{2})$/i.freeze
+  INVALID_POSTCODE = /^(BF|BX|GIR|XX|GY|JE|IM|AI|GX|KY|VG)/i.freeze
+
+  def self.valid?(postcode)
+    postcode = postcode.gsub(/[`~,.<>;':"\/\[\]|{}()=_+-]| /, "")
+    !!(postcode !~ INVALID_POSTCODE && postcode =~ VALID_POSTCODE)
```

Comment:
This would be more readable (and [faster](https://stackoverflow.com/a/42834177)) as:

```suggestion
    postcode.match?(VALID_POSTCODE) && !postcode.match?(INVALID_POSTCODE)
```

Reply:
Looks great, thank you @ChrisBAshton!